contracts/modules/legacy/LegacyGateway.sol

```diff
@@ -3,191 +3,73 @@
+  function quotationData() internal view returns (IQuotationData) {
+    return IQuotationData(internalContracts[uint(ID.QD)]);
   }
 
-  /// @dev Migrates covers from V1 to V2
-  ///
-  /// @param coverId     V1 cover identifier
-  /// @param {bytes}     Additional data that can be passed by Distributor.sol callers
-  function submitClaim(uint coverId, bytes calldata /*data*/) external override returns (uint) {
-    coverMigrator.migrateCoverFrom(coverId, msg.sender, tx.origin);
-    return 0;
+  function memberRoles() internal view returns (IMemberRoles) {
+    return IMemberRoles(internalContracts[uint(ID.MR)]);
   }
 
-  function claimTokens(
-    uint coverId,
-    uint /*incidentId*/,
-    uint coveredTokenAmount,
-    address coveredToken
-  ) external override returns (uint claimId, uint payoutAmount, address payoutToken) {
-    IERC20 token = IERC20(coveredToken);
-    token.transferFrom(msg.sender, address(this), coveredTokenAmount);
-    token.approve(address(incidents), coveredTokenAmount);
-    (claimId, payoutAmount, payoutToken) = incidents.redeemPayoutForMember(
-      coverId,
-      coverId,
-      coveredTokenAmount,
-      msg.sender
-    );
+  function coverMigrator() internal view returns (ICoverMigrator) {
+    return ICoverMigrator(internalContracts[uint(ID.CL)]);
   }
 
-  function getClaimCoverId(uint claimId) public override view returns (uint) {
-    (, uint coverId) = claimsData.getClaimCoverId(claimId);
-    return coverId;
+  function changeDependentContractAddress() external {
+    internalContracts[uint(ID.TK)] = payable(master.tokenAddress());
+    internalContracts[uint(ID.QD)] = master.getLatestAddress("QD");
+    internalContracts[uint(ID.MR)] = master.getLatestAddress("MR");
+    internalContracts[uint(ID.CL)] = master.getLatestAddress("CL");
   }
 
-  function getPayoutOutcome(uint claimId) external override view returns (
-    ClaimStatus status,
-    uint amountPaid,
-    address coverAsset
-  ) {
-    (, uint coverId) = claimsData.getClaimCoverId(claimId);
-    (, uint internalClaimStatus) = claimsData.getClaimStatusNumber(claimId);
-
-    coverAsset = getCurrencyAssetAddress(quotationData.getCurrencyOfCover(coverId));
-    if (internalClaimStatus == 14) {
-      (,address productId) = quotationData.getscAddressOfCover(coverId);
-      address coveredTokenAddress = incidents.coveredToken(productId);
-      if (coveredTokenAddress != address(0)) {
-        amountPaid = incidents.claimPayout(claimId);
-      } else {
-        amountPaid = quotationData.getCoverSumAssured(coverId) * 10 ** assetDecimals(coverAsset);
-      }
-    } else {
-      amountPaid = 0;
-    }
-
-    if (internalClaimStatus == 6 || internalClaimStatus == 9 || internalClaimStatus == 11) {
-      status = ClaimStatus.REJECTED;
-    } else if (internalClaimStatus == 13 || internalClaimStatus == 14) {
-      status = ClaimStatus.ACCEPTED;
-    } else {
-      status = ClaimStatus.IN_PROGRESS;
-    }
+  function initializeDAI() external {
```

Comment:
do we still need this function? DAI is initialized

Reply:
removed